corehq/apps/app_manager/tasks.py

```diff
@@ -1,9 +1,23 @@
 from celery.task import task
 from corehq.apps.users.models import CommCareUser
+from corehq.util.decorators import serial_task
 
 
 @task(queue='background_queue', ignore_result=True)
 def create_user_cases(domain_name):
     from corehq.apps.callcenter.utils import sync_usercase
     for user in CommCareUser.by_domain(domain_name):
         sync_usercase(user)
+
+
+@serial_task('{app._id}', max_retries=0)
```

Comment:
That's true, 5 minutes may be a little low for some domains.  @proteusvacuum do you have any idea how long a reasonable build timeout would be for eNikshay?

Note that there's not a huge issue with an obviously long timeout, as that only triggers when the task hangs indefinitely.